src/Controllers/ElementalAreaController.php

```diff
@@ -29,23 +33,217 @@ class ElementalAreaController extends CMSMain
 
     private static $url_handlers = [
         'elementForm/$ItemID' => 'elementForm',
-        'POST api/saveForm/$ID' => 'apiSaveForm',
         '$FormName/field/$FieldName' => 'formAction',
+        'DELETE api/archive/$ID!' => 'apiArchive',
+        'GET api/readElementalArea/$elementalAreaID!' => 'apiReadElementalArea',
+        'POST api/create' => 'apiCreate',
+        'POST api/duplicate' => 'apiDuplicate',
+        'POST api/publish' => 'apiPublish',
+        'POST api/saveForm/$ID' => 'apiSaveForm',
+        'POST api/sort' => 'apiSort',
+        'POST api/unpublish' => 'apiUnpublish',
     ];
 
     private static $allowed_actions = [
         'elementForm',
-        'apiSaveForm',
         'formAction',
+        'apiCreate',
+        'apiArchive',
+        'apiDuplicate',
+        'apiPublish',
+        'apiReadElementalArea',
+        'apiSaveForm',
+        'apiSort',
+        'apiUnpublish',
     ];
 
+    public function apiArchive(HTTPRequest $request): HTTPResponse
+    {
+        $this->checkJsonCsrfToken($request);
+        $id = (int) $request->param('ID');
+        $element = BaseElement::get()->byID($id);
+        if (!$element) {
+            $this->jsonError(400);
```

Comment:
As I've mentioned in other PRs please add a code comment explaining why it's not 404, if indeed it shouldn't be.

Reply:
As discussed offline, should be 400 because it's a POST request